src/libs/ReportUtils.ts

```diff
@@ -1538,7 +1538,7 @@ function isUnreadWithMention(report: OnyxEntry<Report> | OptionData): boolean {
     // lastMentionedTime and lastReadTime are both datetime strings and can be compared directly
     const lastMentionedTime = report.lastMentionedTime ?? '';
     const lastReadTime = report.lastReadTime ?? '';
-    return lastReadTime < lastMentionedTime;
+    return Boolean('isUnreadWithMention' in report && report.isUnreadWithMention) || lastReadTime < lastMentionedTime;
```

Comment:
Couldn't this just be this: 
```suggestion
    return Boolean(report.isUnreadWithMention) || lastReadTime < lastMentionedTime;
```

Because if report.UnreadWithMention is undefined or null it'll be false? Or will it throw?

Reply:
makes sense! Updated

Reply:
Ah, no, that was added because isUnreadWithMention is not exist in OnyxEntry<Report> and thats a verification to pass TS.
Boolean(report.isUnreadWithMention) - this will cause an error: Property 'isUnreadWithMention' does not exist on type 'Report | OptionData'.
Property 'isUnreadWithMention' does not exist on type 'Report'

Reply:
Ahh ok, I see. Perhaps then, instead of calling the argument `report` we should call it `reportOrOption`. So that it's more clear that the thing we are doing these checks on could be a report or option.